router/routes_test.go

```diff
@@ -16,3 +28,33 @@ func TestRouterGetAll(t *testing.T) {
 		t.Error("expected '[]' got:", routes)
 	}
 }
+
+func TestRouterNoDuplicateIds(t *testing.T) {
+	AdapterFactories.Register(newDummyAdapter, "syslog")
+
+	//Mock "running" so routes actually start running when added.
+	Routes.routing = true
+
+	//Start the first route.
+	route1 := &Route{
+		ID: "abc",
+		Address: "someUrl",
+		Adapter: "syslog",
+	}
+	err := Routes.Add(route1)
```

Comment:
nitpick: you can combine L44 & L45
`if err := Routes.Add(route1); err != nil {`